docs/source/notes/broadcasting.rst

```diff
@@ -86,6 +86,42 @@ For Example::
     >>> (x.add_(y)).size()
     RuntimeError: The expanded size of the tensor (1) must match the existing size (7) at non-singleton dimension 2.
 
+Scatter and gather
+------------------
+
+``gather``, ``scatter_``, ``scatter_add_``, ``scatter``, and ``scatter_add`` also support broadcasting. But the broadcasting
+of these operators are more complicated than most other operators.
+
+When we do ``input.gather(dim, input)``, the following rules are applied:
+
+- The number of dimensions of ``index`` must be equal to or smaller than the number of dimensions of ``input``.
+- If the number of dimensions of ``index`` is smaller than of ``input``, then ``index`` is aligned to the left,
```

Comment:
If we do decide that left alignment is the way to go, it is worth calling out in super bold letters that this is what's happening here, and this is the primary difference between the broadcasting rules in the initial case and post case.

Reply:
Since there is a lot of text here, it's easy to miss